### PR TITLE
DBZ-3295 Restrict all LogMiner results to specified SCN boundary

### PR DESCRIPTION
**Pull request by Naros**
https://issues.redhat.com/browse/DBZ-3295

It was brought to our attention that certain operations such as COMMIT, ROLLBACK, MISSING_SCN, and DDL were not scoped to the defined SCN [start,end] window.  Additionally, the original change for DBZ-2875 in Beta2 would eventually lead to a situation where an event could be missed due to the SCN boundary being used in the LogMiner query.  Both of these have been addressed in this PR which should ultimately conclude changes needed for DBZ-2875.

### CODE REVIEW

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/SqlUtils.java [hunk 1]
```diff
@@ -223,9 +223,14 @@ static String logMinerContentsQuery(OracleConnectorConfig connectorConfig, Strin
...
         query.append("SELECT SCN, SQL_REDO, OPERATION_CODE, TIMESTAMP, XID, CSF, TABLE_NAME, SEG_OWNER, OPERATION, USERNAME ");
         query.append("FROM ").append(LOGMNR_CONTENTS_VIEW).append(" ");
-        query.append("WHERE OPERATION_CODE IN (1,2,3,5) ");
-        query.append("AND SCN > ? ");
-        query.append("AND SCN < ? ");
+        query.append("WHERE ");
+        query.append("SCN > ? AND SCN <= ? ");
+        query.append("AND (");
+        // COMMiT/ROLLBACK/MISSING SCN
+        query.append("(OPERATION_CODE IN (7,34,36) AND USERNAME NOT IN (").append(getExcludedUsers(logMinerUser)).append(")) ");
```
Comment (ckamm) [inline debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/SqlUtils.java:230]: This should change to capture COMMIT and ROLLBACK even from the logMinerUser, see 5d7faf42a926791eef0362c985da2fa8b8eb697f

Comment (Naros) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/SqlUtils.java:230]: Thanks @ckamm, I'll push a commit.  Appreciate the quick review.

**Comment by gunnarmorling** [comment]
@Naros, could you please log a separate issue, reference it in the commit, and close DBZ-2875 again? Issues should not be re-opened after they have been release, so to not bend reality. Thanks!

@jpechane, could you take a look at this one, too?

**Review by ckamm — APPROVED** [review]

**Comment by Naros** [comment]
@gunnarmorling New issue logged & commit message updated as requested.

**Comment by gunnarmorling** [comment]
LGTM; awaiting review by @jpechane, should be good to merge then. Thanks for reviewing to @ckamm, too!

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/SqlUtils.java [hunk 2]
```diff
@@ -223,9 +223,16 @@ static String logMinerContentsQuery(OracleConnectorConfig connectorConfig, Strin
...
-        query.append("AND SCN > ? ");
-        query.append("AND SCN < ? ");
+        query.append("WHERE ");
+        query.append("SCN > ? AND SCN <= ? ");
+        query.append("AND (");
+        // MISSING_SCN/DDL only when not performed by excluded users
+        query.append("(OPERATION_CODE IN (5,34) AND USERNAME NOT IN (").append(getExcludedUsers(logMinerUser)).append(")) ");
+        // COMMIT/ROLLBACK
+        query.append("OR (OPERATION_CODE IN (7,36)) ");
+        // INSERT/UPDATE/DELETE/DDL
```
Comment (jpechane) [inline debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/SqlUtils.java:233]: DDL is probably not part of this predicate so it should be removed form the comment

**Comment by gunnarmorling** [comment]
@Naros, can you check on Jiri's last comment and then merge yourself? Thx!

**Comment by Naros** [comment]
@gunnarmorling Just added a commit to address Jiri's last comment.  Proceeding with merge.

**Pull request merged**